taxi/views.py

```diff
@@ -61,10 +63,28 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def get_queryset(self):
+        return Car.objects.prefetch_related("drivers").all()
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["is_driver"] = self.request.user in self.object.drivers.all()
```

Comment:
So. In order to do "in" operator you call "self.object.drivers.all()". The order of operations is:
1. self.object.drivers.all() - django fetches drivers. They aren't prefetched, thus 2 SQL queries are called.
2. Django fetched all records from db
3. Python runs "in" operator that will go through all records if there is a single object

You can optimize this on ORM lvl so you don't need to use "in" operator. Let's do it
The same for 77 line in "post" method